Source/OxyPlot/PlotModel/PlotModel.Rendering.cs

```diff
@@ -77,7 +77,7 @@ protected virtual void RenderOverride(IRenderContext rc, OxyRect rect)
                         l.EnsureLegendProperties();
                     }
 
-                    while (true)
+                    for (var i = 0; i < 10; i++) // make we sure we don't loop infinitely
```

Comment:
Yes, I did. It seems to very rare, but it happened with one example from the Issues section.
I is also fairly conceivable how this happens:

1. Axis intervals are updated with 0 margins
2. Margin is updated to accommodate tick labels
3. Axis intervals are updated with larger margins, so less space -> different intervals might be generated
4. Margin is updated with new intervals -> by chance they now need less space than before
5. Axis intervals are updated with new, smaller margin -> back to step 2

I'm not sure if there is a more elegant way to prevent this endless loop... Just naively limiting the number of iterations we allow seems to work pretty well.

Reply:
OK. My only thought was that perhaps the axes could 'grow only' but I never did test this when I was trying to work out subplots a while back. It certainly doesn't 'feel' wrong in any of the examples I threw about: do you remember which one caused the problem?

Reply:
Hmm interesting thought, yes maybe we should not allow the margin to shrink, because if we would end up in an endless loop, the larger margin should be the better choice.
No, I don't really remember, but it was something fairly unspectacular where the numbers just randomly aligned such that this loop never finished.

Reply:
I played around a bit, but I think not allowing the margin to shrink is not an adequate solution, as it results in larger margins than necessary in some cases.
I think I'd leave it as is for now and just keep the iteration limit.